src/routes/v1/friendships/mutuals.rs

```diff
@@ -0,0 +1,45 @@
+use actix_web::{
+    get,
+    web::{self, Data},
+    HttpMessage, HttpRequest, HttpResponse,
+};
+
+use super::{errors::FriendshipsError, types::FriendshipsResponse};
+use crate::{
+    components::app::AppComponents, entities::friendships::FriendshipRepositoryImplementation,
+    middlewares::check_auth::UserId, routes::v1::error::CommonError,
+};
+
+#[get("/v1/friendships/{userId}/mutuals")]
+pub async fn get_mutual_friends(
+    req: HttpRequest,
+    user_id: web::Path<String>,
+    app_data: Data<AppComponents>,
+) -> Result<HttpResponse, FriendshipsError> {
+    let logged_in_user = {
+        let extensions = req.extensions();
+        extensions
+            .get::<UserId>()
+            .expect("to have a UserId")
+            .0
+            .clone()
+    };
```

Comment:
Simpler version to read?
```suggestion
    let logged_in_user = req
        .extensions()
        .get::<UserId>()
        .expect("to have a UserId")
        .0
        .clone();
```